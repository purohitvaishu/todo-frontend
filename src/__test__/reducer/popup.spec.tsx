import reducer from "../../reducers/popup.reducers";
import {
  ADD_LIST_POPUP,
  EDIT_LIST_POPUP,
  DELETE_LIST_POPUP,
  GET_LIST_REQUEST,
  GET_LIST_FAILED,
  GET_LIST_SUCCESS,
  ADD_LIST_REQUEST,
  ADD_LIST_FAILED,
  ADD_LIST_SUCCESS
} from "../../actions/popup.actions";
import {
  initialState,
  getList,
  popupList,
  addList
} from "../../mockData/reducer.data";

describe("Popup: reducer", () => {
  test("set initial state", () => {
    expect(reducer(undefined, {})).toMatchSnapshot();
  });

  describe("Popup function", () => {
    test("on request", () => {
      const action = { type: ADD_LIST_POPUP, data: popupList };
      expect(reducer(initialState, action)).toMatchSnapshot();
    });

    test("on success", () => {
      const action = { type: EDIT_LIST_POPUP, data: popupList };
      expect(reducer(initialState, action)).toMatchSnapshot();
    });

    test("on failure", () => {
      const action = { type: DELETE_LIST_POPUP, data: {} };
      expect(reducer(initialState, action)).toMatchSnapshot();
    });
  });

  describe("Get List", () => {
    test("on request", () => {
      const action = { type: GET_LIST_REQUEST, data: {} };
      expect(reducer(initialState, action)).toMatchSnapshot();
    });

    test("on success", () => {
      const action = { type: GET_LIST_SUCCESS, data: getList };
      expect(reducer(initialState, action)).toMatchSnapshot();
    });

    test("on failure", () => {
      const action = { type: GET_LIST_FAILED, data: {} };
      expect(reducer(initialState, action)).toMatchSnapshot();
    });
  });

  describe("Add List", () => {
    test("on request", () => {
      const action = { type: ADD_LIST_REQUEST, data: {} };
      expect(reducer(initialState, action)).toMatchSnapshot();
    });

    test("on success", () => {
      const action = { type: ADD_LIST_SUCCESS, data: addList };
      expect(reducer(initialState, action)).toMatchSnapshot();
    });

    test("on failure", () => {
      const action = { type: ADD_LIST_FAILED, data: {} };
      expect(reducer(initialState, action)).toMatchSnapshot();
    });
  });
});
