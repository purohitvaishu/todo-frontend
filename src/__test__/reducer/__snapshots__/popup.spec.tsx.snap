// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popup: reducer Add List on failure 1`] = `
Object {
  "bucketname": "",
  "completed": false,
  "description": "",
  "edit": false,
  "id": null,
  "list": Array [],
  "open": false,
  "title": "",
}
`;

exports[`Popup: reducer Add List on request 1`] = `
Object {
  "bucketname": "",
  "completed": false,
  "description": "",
  "edit": false,
  "id": null,
  "list": Array [],
  "open": false,
  "title": "",
}
`;

exports[`Popup: reducer Add List on success 1`] = `
Object {
  "bucketname": "",
  "completed": false,
  "description": "",
  "edit": false,
  "id": null,
  "list": Array [
    Object {
      "bucketname": "Grocery",
      "data": Array [
        Object {
          "completed": false,
          "description": "For every morning 1ltr",
          "id": 6,
          "title": "Milk",
        },
      ],
    },
  ],
  "open": false,
  "title": "",
}
`;

exports[`Popup: reducer Get List on failure 1`] = `
Object {
  "bucketname": "",
  "completed": false,
  "description": "",
  "edit": false,
  "id": null,
  "list": Array [],
  "open": false,
  "title": "",
}
`;

exports[`Popup: reducer Get List on request 1`] = `
Object {
  "bucketname": "",
  "completed": false,
  "description": "",
  "edit": false,
  "id": null,
  "list": Array [],
  "open": false,
  "title": "",
}
`;

exports[`Popup: reducer Get List on success 1`] = `
Object {
  "bucketname": "",
  "completed": false,
  "description": "",
  "edit": false,
  "id": null,
  "list": Array [
    Object {
      "bucketname": "Trip to Paris",
      "data": Array [
        Object {
          "completed": true,
          "description": "Check passport for migrations",
          "id": 2,
          "title": "Passport Check",
        },
      ],
    },
  ],
  "open": false,
  "title": "",
}
`;

exports[`Popup: reducer Popup function on failure 1`] = `
Object {
  "bucketname": "",
  "completed": false,
  "description": "",
  "edit": false,
  "id": null,
  "list": Array [],
  "open": false,
  "title": "",
}
`;

exports[`Popup: reducer Popup function on request 1`] = `
Object {
  "bucketname": "The Godfather",
  "completed": false,
  "description": "Submit to doctor",
  "edit": true,
  "id": 5,
  "list": Array [],
  "open": true,
  "title": "File",
}
`;

exports[`Popup: reducer Popup function on success 1`] = `
Object {
  "bucketname": "",
  "completed": false,
  "description": "",
  "edit": false,
  "id": null,
  "list": Array [],
  "open": false,
  "title": "",
}
`;

exports[`Popup: reducer set initial state 1`] = `
Object {
  "bucketname": "",
  "completed": false,
  "description": "",
  "edit": false,
  "id": null,
  "list": Array [],
  "open": false,
  "title": "",
}
`;
